File: ChickenEgg.cpp
#include "ChickenEgg.hpp"

//Constructor menerima harga, dan mempassing producer untuk superclass.
ChickenEgg::ChickenEgg(int _price, string _producer){
	initializeFarmProduct(_producer);
	initializeProduct(_price, FARMPRODUCT, CHICKEN_EGG, '?');
}

File: Java/src/main/java/engisubs/gameobject/product/farm_product/egg/ChickenEgg.java
package main.java.engisubs.gameobject.product.farm_product.egg;

public class ChickenEgg extends Egg{
    public ChickenEgg(int _price, String _producer){
        super.initializeFarmProduct(_producer);
        super.initializeProduct(_price, ProductType.FARMPRODUCT, GameObjectType.CHICKEN_EGG, "Telur Ayam Pasaran");
    }
}